### PR TITLE
Suppress all compilation warnings

### PR DESCRIPTION
**Pull request by scc-tw**

### CODE REVIEW

File: src/template.c [hunk 1]
```diff
@@ -50,7 +50,7 @@ char *templateRender(Template *template)
     rewind(file);
 
     buff = malloc(sizeof(char) * (len + 2));
-    fread(buff + 1, sizeof(char), len, file);
+    (void)!fread(buff + 1, sizeof(char), len, file);
```
Comment (jserv) [inline src/template.c]: I dislike the way to introduce a logical NOT operator here. Can you use alternative ones?

Comment (jserv) [reply src/template.c]: Any progress so far?

Comment (scc-tw) [reply src/template.c]: > Any progress so far?

Update GCC or change the compiler would dismiss this warning message,
(I got that warning at GCC 9.1, but it disappears at GCC 12.1 now)

**Comment by jserv** [comment]
Thank @25077667 for contributing!

**Pull request merged**